From 6e3442b033526813c1810a09816273a4f1e1fd42 Mon Sep 17 00:00:00 2001
From: Thierry Escande <thierry.escande@vates.tech>
Date: Tue, 24 Oct 2023 17:19:12 +0000
Subject: [PATCH] Fix varstore-sb-state exit code

In 'user' mode, varstore-sb-state exit code is the returned value of the
function setup_keys() which returns true on success and false otherwise.
The exit code is then 1 on success and 0 on failure, the opposite of any
command line tool exit code.

This patch fixes the exit code, returning 0 when setup_keys() succeeds
and 1 otherwise.

Signed-off-by: Thierry Escande <thierry.escande@vates.tech>
---
 tools/varstore-sb-state.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/varstore-sb-state.c b/tools/varstore-sb-state.c
index 6305dd2..30a5f66 100644
--- a/tools/varstore-sb-state.c
+++ b/tools/varstore-sb-state.c
@@ -112,7 +112,7 @@ int main(int argc, char **argv)
     do_rm(&gEfiImageSecurityDatabaseGuid, "dbx");
 
     if (!strcmp(argv[optind + 1], "user"))
-        return setup_keys();
+        return (setup_keys() ? 0 : 1);
     else
         return 0;
 }
